src/plugins/mer/meremulatordevice.cpp

```diff
@@ -737,6 +819,37 @@ void MerEmulatorDeviceManager::onDeviceListReplaced()
             MerVirtualBoxManager::updateEmulatorQmlLivePorts(merEmulator->virtualMachine(),
                     merEmulator->qmlLivePortsList());
         m_deviceQmlLivePortsCache.insert(merEmulator->id(), nowQmlLivePorts);
+
+        const int nowMemorySize = merEmulator->memorySizeMb();
+        if (nowMemorySize != oldMemorySizeCache.value(merEmulator->id()))
+            MerVirtualBoxManager::setMemorySizeMb(merEmulator->virtualMachine(), nowMemorySize);
+
+        m_deviceMemorySizeCache.insert(merEmulator->id(), nowMemorySize);
+
+        const int nowCpuCount = merEmulator->cpuCount();
+        if (nowCpuCount != oldCpuCountCache.value(merEmulator->id()))
+            MerVirtualBoxManager::setCpuCount(merEmulator->virtualMachine(), nowCpuCount);
+
+        m_deviceCpuCountCache.insert(merEmulator->id(), nowCpuCount);
+
+        const int nowVdiCapacityMb = merEmulator->vdiCapacityMb();
+        if (nowVdiCapacityMb != oldVdiInfoCache.value(merEmulator->id())) {
+            QEventLoop loop;
+            bool needLoop = true;
+
+            MerVirtualBoxManager::setVdiCapacityMb(merEmulator->virtualMachine(), nowVdiCapacityMb, &loop, [&loop, &needLoop] (bool ok) {
+                Q_UNUSED(ok);
+                if (!loop.isRunning())
+                    needLoop = false;
+                else
+                    loop.exit();
+            });
+
+            if (needLoop) {
+                loop.exec();
```

Comment:
Handle failure?

Reply:
When apply emulator settings the error is not processed anywhere. Add it?

Reply:
Now I noticed it is not handled in other cases either, so let's forget about.